client/api/src/execution_extensions.rs

```diff
@@ -165,6 +168,7 @@ impl<T: offchain::DbExternalities + Clone + Sync + Send + 'static> DbExternaliti
 pub struct ExecutionExtensions<Block: BlockT> {
 	strategies: ExecutionStrategies,
 	keystore: Option<KeystorePtr>,
+	mixnet_kx_public_store: Option<Arc<MixnetKxPublicStore>>,
```

Comment:
I didn't think the requirements overlapped enough for this to make sense. The existing keystore is (AFAICT) designed for long-lived keys that persist across node restarts. The mixnet key-exchange keys are only used for a single session (a few hours), and should be destroyed once the session ends. They intentionally do not persist across node restarts, as this would be non-trivial to implement properly:

- When destroying keys you would have to be careful to properly erase the key data from disk.
- The corresponding replay filters would also have to persist, or you would be vulnerable to replay attacks after a node restart.

I opened a GitHub issue about this a while back (#12595), there wasn't really any conclusion though.